# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


In [ ]:
import numpy as np
from scipy import stats

In [ ]:
times =np.array ([128, 135, 121, 142, 126,151, 123])
mu0=123.7
alpha=0.10

In [ ]:
#Крок 1. Формування гіпотез

H₀: μ = 123.7 хв   
H₁: μ > 123.7 хв  
Тип тесту: правосторонній t-тест

In [ ]:
#Крок 2.Обчисліть вибіркові статистики

In [ ]:
x_bar=np.mean(times)
s=np.std(times)
n=len(times)

In [ ]:
#Крок 3. Обчислюємо критичне значення, статистику тесту та р-значення

In [ ]:
res=stats.ttest_1samp(times,popmean=mu0, alternative='greater')
print(f"scipy:t= {res.statistic:.3f}, p_value= {res.pvalue:.5f}")

scipy:t= 2.069, p_value= 0.04202


In [ ]:
res

TtestResult(statistic=np.float64(2.0687273144186475), pvalue=np.float64(0.042017654253682576), df=np.int64(6))

In [ ]:
df=n-1

In [ ]:
t_crit = stats.t.ppf(alpha, df)
t_crit

np.float64(-1.4397557472577691)

In [ ]:
#Крок 4. Висновок на основі p-значення

In [ ]:
if res.pvalue < alpha:
    print(f"Відхиляємо H0: середній час до покупки збільшився на рівні значущості {alpha}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Відхиляємо H0: середній час до покупки збільшився на рівні значущості 0.1


In [ ]:
for alpha in [0.10, 0.05]:
    decision = "Відхиляємо H0" if res.pvalue <= alpha else "Не відхиляємо H0"
    print(f"alpha={alpha:.2f}: рішення → {decision}")

alpha=0.10: рішення → Відхиляємо H0
alpha=0.05: рішення → Відхиляємо H0


**Інтерпретація результату:**
Статистично виявлено, що справді редизайн сайту подовжив час до покупки



**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [ ]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

count = 311   # кількість "успіхів"
nobs = 1200    # обсяг вибірки
p0 = 0.23      # гіпотетична пропорція (H0)

# z-тест для пропорції
stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha=0.03, method='normal')

print("🔹 Тест для однієї пропорції")
print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"95% довірчий інтервал: {confint}")

🔹 Тест для однієї пропорції
Z-статистика = 2.306
p-value = 0.0106
95% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


Інтерпретація результати

1. Односторонній тест

   * $H_0: p = 0.23$
   * $H_1: p > 0.23$
   * Оскільки $p\text{-value} = 0.0106 < 0.03$, відхиляємо $H_0$.
     👉 Є статистично значущі докази, що відсоток тих, хто впізнає цей логотип, **більший за 23%**.

2. 97% довірчий інтервал (0.231, 0.286)

   * Це означає, що з імовірністю 97% справжня частка лежить у межах **від 23.1% до 28.6%**.
   